web/src/narrow.js

```diff
@@ -92,6 +92,10 @@ export function save_narrow(terms, trigger) {
     }
     const new_hash = hash_util.search_terms_to_hash(terms);
     changehash(new_hash, trigger);
+
+    if (stream_list.is_zoomed_in()) {
+        history.replaceState({show_more_topics: true}, "", window.location.href);
```

Comment:
Does this blow away any value for the other `narrow_opts` that might be present?

Reply:
changehash set the state to null. So there isn't narrow_opts available at this stage.

Reply:
Added a common function ` update_current_history_state_data()` to "just takes the existing history.state object, if any, and sets the show_more_topics field on it."